mmdet/models/backbones/vit.py

```diff
@@ -0,0 +1,529 @@
+            trunc_normal_(m.weight, std=.02)
+            if isinstance(m, nn.Linear) and m.bias is not None:
+                nn.init.constant_(m.bias, 0)
+        elif isinstance(m, nn.LayerNorm) or isinstance(m, nn.BatchNorm2d):
+            nn.init.constant_(m.bias, 0)
+            nn.init.constant_(m.weight, 1.0)
+        elif isinstance(m, nn.Conv2d) or isinstance(m, nn.ConvTranspose2d):
+            fan_out = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
+            fan_out //= m.groups
+            m.weight.data.normal_(0, math.sqrt(2.0 / fan_out))
+            if m.bias is not None:
+                m.bias.data.zero_()
+
+    def forward_features(self, x):
+        B = x.shape[0]
+        x, patch_resolution = self.patch_embed(x)
+
+        # stole cls_tokens impl from Phil Wang, thanks
+        cls_tokens = self.cls_token.expand(B, -1, -1)
+        x = torch.cat((cls_tokens, x), dim=1)
+        x = x + resize_pos_embed(
+            self.pos_embed,
+            self.patch_resolution,
+            patch_resolution,
+            mode=self.interpolate_mode,
+            num_extra_tokens=self.num_extra_tokens)
+        x = self.drop_after_pos(x)
+
+        if not self.with_cls_token:
+            # Remove class token for transformer encoder input
+            x = x[:, 1:]
+
+        outs = []
+        for i, layer in enumerate(self.layers):
+            rel_pos_bias = self.local_share_rel_pos_bias() if i not in \
+                    self.out_indices else self.global_share_rel_pos_bias()
+            x = layer(x, patch_resolution[0], patch_resolution[1],
+                      rel_pos_bias[:, 1:, 1:])
+
+            if i == len(self.layers) - 1 and self.final_norm:
+                x = self.norm1(x)
+
+            if i in self.out_indices:
+                B, _, C = x.shape
+                if self.with_cls_token:
+                    patch_token = x[:, 1:]
+                    cls_token = x[:, 0]
+                else:
+                    patch_token = x
+                    cls_token = None
+                if self.output_cls_token:
+                    out = [patch_token, cls_token]
+                else:
+                    out = patch_token
+                outs.append(out)
+
+        return tuple(outs), patch_resolution
+
+    def forward(self, x):
+        # Rescale feature maps to multi-scale
+        outs, patch_resolution = self.forward_features(x)
+        B, _, C = outs[0].shape
+
+        # Normalization and Reshape
+        out0 = self.norm0(outs[0]).transpose(1, 2).reshape(
+            B, C, *patch_resolution)
+        out1 = self.norm1(outs[1]).transpose(1, 2).reshape(
+            B, C, *patch_resolution)
+        out2 = self.norm2(outs[2]).transpose(1, 2).reshape(
+            B, C, *patch_resolution)
+        out3 = self.norm3(outs[3]).transpose(1, 2).reshape(
+            B, C, *patch_resolution)
+
+        # Convert to multi-scale
+        out0 = self.res_modify_block_0(out0).contiguous()
```

Comment:
create a channel mapper module to map the feature maps to the expected shape and channels.